test/test_optim.py

```diff
@@ -1493,5 +1493,31 @@ def test_cosine_then_cyclic(self):
 
         self.assertLessEqual(last_lr, max_lr)
 
+    def test_onecycle_lr_div_factors(self):
+        # https://github.com/pytorch/pytorch/issues/28216
+
+        model = torch.nn.Linear(2, 1)
+        optimizer = torch.optim.SGD(model.parameters(), lr=0.5)
+        div_factors = [1, 10, 25, 100]
+        final_div_factors = [1, 1E2, 1E4, 1E6]
+
+        for div_factor, final_div_factor in zip(div_factors, final_div_factors):
+            for strategy in ['linear', 'cos']:
+                scheduler = OneCycleLR(optimizer, max_lr=25, total_steps=10,
```

Comment:
I'd expect to have a new optimizer created before this line, and your test to fail without this PR. Is this the case?

Reply:
If the optimizer is created before line 1506, then my test will pass even without the pull request. The way the test is written it will pass with the pull request, but fail without.

Reply:
Right, the code change looks good to me then, but not the test :) We should not encourage users in experimenting with new schedulers without creating new optimizers. It would have been nice to have a different use case that would fail.

The change the PR proposes though is in line with the intended use since `initial_lr` is used in `step`.

Now that the issue has been clarified, let's remove the test. I'll approve and merge the code without it. Thanks for looking into this!

Reply:
Sure. Thanks for the clarifications.